imagekit-sdk/src/test/java/io/imagekit/sdk/ImageKitTest.java

```diff
@@ -2,629 +2,1890 @@
+			throws InterruptedException, ForbiddenException, TooManyRequestsException, InternalServerException,
+			UnauthorizedException, BadRequestException, UnknownException, IOException {
+
+		MockWebServer server = new MockWebServer();
+		String responseJson = "[\n" + "    {\n" + "        \"type\": \"file\",\n"
+				+ "        \"name\": \"default-image.jpg\",\n"
+				+ "        \"createdAt\": \"2022-06-11T07:26:19.294Z\",\n"
+				+ "        \"updatedAt\": \"2022-06-11T07:26:19.600Z\",\n"
+				+ "        \"fileId\": \"62a4439bce686814dfcce65c\",\n" + "        \"tags\": null,\n"
+				+ "        \"AITags\": null,\n" + "        \"versionInfo\": {\n"
+				+ "            \"id\": \"62a4439bce686814dfcce65c\",\n" + "            \"name\": \"Version 1\"\n"
+				+ "        },\n" + "        \"embeddedMetadata\": {\n"
+				+ "            \"DateCreated\": \"2022-06-11T07:26:19.599Z\",\n"
+				+ "            \"DateTimeCreated\": \"2022-06-11T07:26:19.600Z\"\n" + "        },\n"
+				+ "        \"customCoordinates\": null,\n" + "        \"customMetadata\": {},\n"
+				+ "        \"isPrivateFile\": false,\n"
+				+ "        \"url\": \"https://ik.imagekit.io/zv3rkhsym/default-image.jpg\",\n"
+				+ "        \"thumbnail\": \"https://ik.imagekit.io/zv3rkhsym/tr:n-ik_ml_thumbnail/default-image.jpg\",\n"
+				+ "        \"fileType\": \"image\",\n" + "        \"filePath\": \"/default-image.jpg\",\n"
+				+ "        \"height\": 1000,\n" + "        \"width\": 1000,\n" + "        \"size\": 147022,\n"
+				+ "        \"hasAlpha\": false,\n" + "        \"mime\": \"image/jpeg\"\n" + "    }\n" + "]";
+		server.enqueue(new MockResponse().setBody(responseJson));
+		server.start();
+		RestClient.API_BASE_URL = server.url("/").toString();
+		Map<String, String> options = new HashMap<>();
+		options.put("limit", "" + 1);
+		SUT.getFileList(options);
+
+		RecordedRequest request = server.takeRequest();
+		String utf8RequestBody = request.getBody().readUtf8();
+		assertEquals("", utf8RequestBody);
+		assertEquals("application/json", request.getHeader("Content-Type"));
+		assertEquals("GET /v1/files?limit=1 HTTP/1.1", request.getRequestLine());
+		assertEquals(RestClient.API_BASE_URL.concat("v1/files?limit=1"), request.getRequestUrl().toString());
+
+	}
+
+	@Test(expected = NotFoundException.class)
+	public void imageKit_getFileDetail__expected_404()
+			throws IOException, InterruptedException, BadRequestException, InternalServerException, UnknownException,
+			ForbiddenException, TooManyRequestsException, UnauthorizedException {
+
+		MockWebServer server = new MockWebServer();
+		String responseJson = "{\n" + "    \"message\": \"The requested file does not exist.\",\n"
+				+ "    \"help\": \"For support kindly contact us at support@imagekit.io .\"\n" + "}";
+		server.enqueue(new MockResponse().setResponseCode(404).setBody(responseJson));
+		server.start();
+		RestClient.API_BASE_URL = server.url("/").toString();
+		SUT.getFileDetail("629f3de17eb0fe4053615450");
```

Comment:
What are we asserting in this test case?

Reply:
@imagekitio the assert is on line with (expected = NotFoundException.class)